prql-compiler/src/parser.rs

```diff
@@ -2041,4 +2055,46 @@ join `my-proj`.`dataset`.`table`
                     named_args: {}
         "###)
     }
+
+    #[test]
+    fn check_valid_version() {
+        let stmt = format!("prql version:{}\n", env!("CARGO_PKG_VERSION_MAJOR"));
+        assert!(parse(&stmt).is_ok());
+
+        let stmt = format!(
+            r#"
+        prql version:"{}"
+        "#,
+            env!("CARGO_PKG_VERSION_MAJOR")
+        );
+        assert!(parse(&stmt).is_ok());
+
+        let stmt = format!(
+            r#"
+            prql version:"{}.{}"
```

Comment:
Do we want this to be a string (i.e. `version:"1.0"), or just bare `version:1.0`. 

It's currently bare.

But if we're doing full semver, potentially `1.0.0` will not be a valid value

Reply:
there are a few way we can go about that I think:
- Make the version a syntax element, I don't like that, because it causes the syntax to become ambiguous
- Current state: accept integer or string that parses into version, when an int is passed, interpret it as a major
- Current version + support for floats, so we also support `major.minor` versions as plain numbers

Reply:
I also don't like having a new syntax just for semver, but also don't like it to be parsed as float, because it's not a float (`1.10 != 1.1`).

So I'd argue against having bare `version:1.0` or `version:1.0.0`. I'm partial about shortcut `version:1`.

Reply:
Do you suggest that we just go full string? This is probably the sanest solution, but it break backward compat

Reply:
It would be the sanest option, true. I doubt that anyone is using this yet so don't worry about compat.

We can add int shortcut/special version syntax element later, if there is pressure to make it more elegant.

Reply:
Yeah, I also think backward compat is much of a concern at this stage.

I wish it could work without needing to quote the strings but I don't know enough about the parser grammar so happy to go with whatever you guys decide.

Reply:
We could add it to the grammar, but it would be ambiguous with floats, so we'd have to treat the whole version:xxx differently, to disambiguate

Reply:
Sorry about the typo, what I meant to say is I think backward compatibility is NOT much of a concern at this stage.

---
Right, your full string proposal sounds like the best then.

Reply:
> Do you suggest that we just go full string? This is probably the sanest solution, but it break backward compat

Yeah, I think at this stage we can be flexible, and make a guess if it's going to break anyone, and I would guess the same as others, that's it's fine. (And we already break things, because we're going to `0`!)



> We can add int shortcut/special version syntax element later, if there is pressure to make it more elegant.

👍